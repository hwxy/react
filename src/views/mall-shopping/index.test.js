
import React from 'react';
import { render, shallow } from 'enzyme';
import TodoList from './index';
import renderer from "react-test-renderer"


const props = {
  list: ['first', 'second'],
  deleteTodo: jest.fn(),
};

const setup = () => {
  const wrapper = shallow(<TodoList {...props} />);
  return {
    props,
    wrapper,
  };
};

const setupByRender = () => {
  const wrapper = render(<TodoList {...props} />);
  return {
    props,
    wrapper,
  };
};

const setupByMount = () => {
  const wrapper = mount(<TodoList {...props} />);
  return {
    props,
    wrapper,
  };
};


it('renders correctly', () => {
  setup();
  console.log(TodoList, 22);
  const tree = renderer.create(<TodoList {...props} />)
          .toJSON();

  expect(tree).toMatchSnapshot();
});
