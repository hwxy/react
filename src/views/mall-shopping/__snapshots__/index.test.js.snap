// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="todo-list"
>
  <div>
    <span
      className="item-text "
    >
      first
    </span>
    <a
      aria-disabled={false}
      className="am-button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
    >
      <span>
        done
      </span>
    </a>
  </div>
  <div>
    <span
      className="item-text "
    >
      second
    </span>
    <a
      aria-disabled={false}
      className="am-button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
    >
      <span>
        done
      </span>
    </a>
  </div>
</div>
`;
